### PR TITLE
Tighten the docs and tests around `site_url`

### PR DESCRIPTION
**Pull request by oprypin**
Some of the docs were accidentally reverted with other things, so I'm restoring them and adding a bit more.

Fixing an edge case: it is in fact possible to specify a URL with scheme and without netloc. Adding new tests to also catch that.

### CODE REVIEW

File: mkdocs/tests/config/config_options_tests.py [hunk 1]
```diff
@@ -280,39 +280,34 @@ def test_invalid_IPv6_address(self):
...
                           option.validate, 1)
 
-    def test_url_is_dir(self):
-        url = "https://mkdocs.org/"
-
-        option = config_options.URL(is_dir=True)
-        value = option.validate(url)
-        self.assertEqual(value, url)
-
-    def test_url_transform_to_dir(self):
```
Comment (ultrabug) [inline mkdocs/tests/config/config_options_tests.py:309]: It looks like what this test was covering is not checked anymore if you remove that test or I might be missing something?

Comment (oprypin) [reply mkdocs/tests/config/config_options_tests.py:309]: The line `self.assertEqual(option.validate("https://mkdocs.org"), "https://mkdocs.org/")` covers exactly this

**Review by ultrabug — APPROVED** [review]
